clients/src/main/java/org/apache/kafka/clients/consumer/KafkaConsumer.java

```diff
@@ -2211,4 +2219,20 @@ private void throwIfNoAssignorsConfigured() {
                 ConsumerConfig.PARTITION_ASSIGNMENT_STRATEGY_CONFIG + " configuration property");
     }
 
+    private static ApiVersions createApiVersions(boolean autoTopicCreationAllowed) {
+        return new ApiVersions(new ApiVersions.ChangeListener() {
+            @Override
+            public void onUpdate(String nodeId, NodeApiVersions nodeApiVersions) {
+                if (!autoTopicCreationAllowed) {
+                    if (!MetadataRequest.disableAutoTopicCreateSupported(nodeApiVersions.apiVersion(ApiKeys.METADATA).maxVersion))
+                        throw new InvalidConfigurationException("Configuration " + ConsumerConfig.ALLOW_AUTO_CREATE_TOPICS_CONFIG +
```

Comment:
Hmm.. I am not sure this will always be raised to a useful place or that it is safe to raise it from `NetworkClient` during response handling. Wouldn't that mean we lose other responses? It feels a little brittle.

A similar use case is the handling of authentication errors. These come before the first request and also need to be propagated back to the user. There might be an opportunity to generalize the authentication error handling so that it can be used for unsatisfied api version requirements as well. It's not the greatest pattern either, but the code paths are mostly known and then we wouldn't need multiple mechanisms to propagate exceptions from the network layer back to the user.